kernel/kernel-api/src/main/java/io/delta/kernel/Snapshot.java

```diff
@@ -29,13 +29,29 @@
 public interface Snapshot {
 
   /**
-   * Get the version of this snapshot in the table.
+   * Get the version of this snapshot in the table. Returns -1 if the table is empty and so has no
+   * commits.
    *
    * @param engine {@link Engine} instance to use in Delta Kernel.
    * @return version of this snapshot in the Delta table
    */
   long getVersion(Engine engine);
 
+  /**
+   * Get the timestamp (in milliseconds since the Unix epoch) of the latest commit of this snapshot.
+   * Returns -1 if the table is empty and so has no commits.
```

Comment:
Maybe instead this should just be undefined for an empty table with no commits? Since -1 is technically a valid timestamp

Reply:
I've changed my mind. Users should never get a Snapshot reference for Delta Table that doesn't exist (i.e. version = -1, or timestamp = -1).

Internal Kernel code may get such a Snapshot reference (e.g. TransactionBuilderImpl creates an InitialSnapshot), but this will never be propagated to users.

So, I can remove this part of the public documentation.

Reply:
I wonder if this might change once we support read-->write transactions?